### DISCUSSION TITLE
Gemini API usage suspended in Frigate Beta 3

### DISCUSSION THREAD

**Discussion by bmrowe** [blakeblackshear/frigate#15582] [Ask A Question]
I'm using the Gemini API in Frigate Beta 3. I noticed in Frigate logs some rate limit exceeded errors being returned by Gemini. I logged into Gemini to check on my limits/quota and saw this:

Resources associated with your project Gemini API (id: gen-lang-client-xxxx) are being suspended for using Gemini API in a manner that violates the applicable[ Gemini API Additional Terms of Service](https://ai.google.dev/gemini-api/terms) as well as the [Google APIs Terms of Service](https://developers.google.com/terms) and [Generative AI Prohibited Use Policy](https://policies.google.com/terms/generative-ai/use-policy).

Has anyone seen this? Is there a better API to be using? I saw in the documentation that I should be getting 15 API calls/minute for free, but its not clear to me when Frigate is calling this API.

**Comment by NickM-27** [comment]
We have not seen any other reports of this, does it indicate what part of the ToS might have been violated?

**Reply by bmrowe** [reply]
It does not. I opened an appeal. I also don't see much in the dev portal, but I'm much more familiar with AWS, so maybe I'm missing something.

**Reply by bmrowe** [reply]
I see this from a traffic perspective:

![image](https://github.com/user-attachments/assets/e245b4c0-9407-4bae-928a-a640c7cc2455)

**Reply by llburg** [reply]
I got flagged as well, my best guess is it's under the "Prohibited Use Policy" "1. g. :Tracks or monitors people without their consent."

**Reply by bmrowe** [reply]
> I got flagged as well, my best guess is it's under the "Prohibited Use Policy" "1. g. :Tracks or monitors people without their consent."

Thanks for chiming in. Did you have any luck appealing?

**Reply by bmrowe** [reply]
@NickM-27 Google removed the suspension after I appealed. Based on what I can tell, some combination of the prompt and images being sent to their API triggered this suspension. I'd advise some testing in this area to prevent other users from encountering similar -- especially given @llburg 's experience as well.

**Reply by hawkeye217** [reply]
There really isn't anything we can do directly about this. The default prompt doesn't even make use of "surveillance", "security camera", "monitor" or "track":

> Analyze the sequence of images containing the {label}. Focus on the likely intent or behavior of the {label} based on its actions and movement, rather than describing its appearance or the surroundings. Consider what the {label} is doing, why, and what it might do next.

Users should be aware of the terms of service for their selected GenAI provider and are free to customize the prompts to their liking: https://deploy-preview-13787--frigate-docs.netlify.app/configuration/genai#custom-prompts

**Reply by ehn** [reply]
Google has [removed its pledge](https://www.cnbc.com/2025/02/04/google-removes-pledge-to-not-use-ai-for-weapons-surveillance.html) to not use AI for weapons and, more interestingly for us, surveillance. Perhaps they will be more open to using Gemini models to analyze surveillance video in the future.

**Comment by leccelecce** [comment]
Slight side-track but attempting to use local Ollama with some models, I've noticed I get the following responses when trying to regenerate images:

`I can’t assist you in describing people for security purposes. Is there something else I can help you with?`
`I can’t provide information that could be used to surveil individuals. Is there anything else I can help you with?`
`I can’t help you with this task as it involves surveillance images. Is there anything else I can help you with?`

This was after I customised the prompt to give it context that the image was from a security camera located on a residential property. Are you using a similar prompt?

**Reply by bmrowe** [reply]
I have not written a custom prompt. I do wonder if the default prompt is what is making Gemini suspend the account

**Comment by primevisionservices** [comment]
I had some problems, but in my case it was using the same project for both the home assistant and the frigate (same key) . For everyone who is going to use Gemini on the frigate, create a new project.

**Comment by JoshuaSeidel** [comment]
use ollama if you have the processing power. it works fast with a 1660 super gpu in my case, and you get no restrictions.

**Comment by BlackBeltPanda** [comment]
Just ran into this, as well:
![image](https://github.com/user-attachments/assets/f5f564ee-4977-4a4d-a232-95d59cf47eb4)

**Reply by Thorsten1982** [reply]
Where exactly did you find this information? I wanted to take a look too.

**Comment by bmrowe** [comment]
I've appealed this and Google removed the suspension. It re-suspended me a couple more times and I had to appeal those as well. I haven't been suspended in awhile, however I do see maybe a dozen gemini timeouts per day so maybe they are throttling me or something:

Gemini returned an error: 503 GOAWAY received; Error code: 0; Debug Text: session_timed_out

**Reply by thelovemyth** [reply]
I have that err too
I hope we'll see more affordable (or free) AI APIs available soon.
